sympy/solvers/tests/test_solvers.py

```diff
@@ -1613,6 +1616,26 @@ def count(dic):
         == 3
     assert count(solve([x + y + z, y + z + a], particular=True, quick=True)) == 1
     assert count(solve([x + y + z, y + z + a], particular=True, quick=False)) == 2
+    # issue 22718
+    A = Matrix([
+        [ 1,  1,  1,  0,  1,  1,  0,  1,  0,  0,  1,  1,  1,  0],
+        [ 1,  1,  0,  1,  1,  0,  1,  0,  1,  0, -1, -1,  0,  0],
+        [-1, -1,  0,  0, -1,  0,  0,  0,  0,  0,  1,  1,  0,  1],
+        [ 1,  0,  1,  1,  0,  1,  1,  0,  0,  1, -1,  0, -1,  0],
+        [-1,  0, -1,  0,  0, -1,  0,  0,  0,  0,  1,  0,  1,  1],
+        [-1,  0,  0, -1,  0,  0, -1,  0,  0,  0, -1,  0,  0, -1],
+        [ 0,  1,  1,  1,  0,  0,  0,  1,  1,  1,  0, -1, -1,  0],
+        [ 0, -1, -1,  0,  0,  0,  0, -1,  0,  0,  0,  1,  1,  1],
+        [ 0, -1,  0, -1,  0,  0,  0,  0, -1,  0,  0, -1,  0, -1],
+        [ 0,  0, -1, -1,  0,  0,  0,  0,  0, -1,  0,  0, -1, -1],
+        [ 0,  0,  0,  0,  1,  1,  1,  1,  1,  1,  0,  0,  0,  0],
+        [ 0,  0,  0,  0, -1, -1,  0, -1,  0,  0,  0,  0,  0,  0]])
+    v = Matrix(symbols("v:14", integer=True))
+    B = Matrix([[2], [-2], [0], [0], [0], [0], [0], [0], [0],
+        [0], [0], [0]])
+    eqs = A@v-B
+    assert solve(eqs) == []
```

Comment:
Why does `solve(eqs)` return no solutions here?

This is the output from `linsolve`:
```python
In [14]: linsolve((A, B))
Out[14]: 
⎧⎛          3⋅τ₂   1                                          τ₂   2          τ₂   4               
⎨⎜τ₀ + τ₁ + ──── + ─, 4/3 - τ₀, -τ₁ - 2/3, -τ₂ - 2/3, -2⋅τ₁ - ── - ─, -2⋅τ₀ - ── + ─, -2/3, 2⋅τ₀ + 
⎩⎝           4     3                                          2    3          2    3               

                                       3⋅τ₂                ⎞⎫
2⋅τ₁ + τ₂ - 2/3, -2/3, 4/3, -τ₀ - τ₁ - ──── + 1, τ₀, τ₁, τ₂⎟⎬
                                        4                  ⎠⎭

```

Reply:
`solve` doesn't handle infinite solutions: the `tau` parameters are showing a parametric solution for the equations.

Reply:
Usually `solve` does handle infinite solutions for linear and polynomial systems:
```python
In [8]: solve([x + y], [x, y])
Out[8]: {x: -y}

In [9]: solve([x + 2 * y + 3 * z, 4 * x + 5 * y + 6 * z, 7 * x + 8 * y + 9 * z], [x, y, z])
Out[9]: {x: z, y: -2⋅z}

In [10]: solve([x ** 2 + y ** 2 + z, z + z], [x, y, z])
Out[10]: [(-ⅈ⋅y, y, 0), (ⅈ⋅y, y, 0)]
```

Reply:
There must be some problem with checking somewhere:
```python
In [22]: sol = solve(eqs, check=False)

In [23]: sol
Out[23]: 
⎧      v₃   v₇   1        v₃   v₅   1       v₃   v₇   5       v₃   v₅   5       v₃   v₅   v₇   1   
⎨v₀: - ── + ── + ─, v₁: - ── + ── + ─, v₁₀: ── - ── + ─, v₁₁: ── - ── + ─, v₁₂: ── + ── + ── - ─, v
⎩      4    2    2        4    2    2       4    2    6       4    2    6       4    2    2    6   

                     v₃   v₅   v₇   1                                           ⎫
₁₃: -v₃ - 2/3, v₂: - ── - ── - ── - ─, v₄: -v₅ - v₇, v₆: -2/3, v₈: -2/3, v₉: 4/3⎬
                     4    2    2    2                                           ⎭

In [24]: eqs.subs(sol).is_zero_matrix
Out[24]: True

In [25]: checksol(eqs, sol)
Out[25]: True
```

Reply:
The `integer` assumption was used but `v1 = 4/3 - v11` so the assumption is violated.

Reply:
When this set of equations is used in master, an error is [raised](https://github.com/sympy/sympy/issues/22718#issue-1084286569), so the test is just checking that the error is no longer there. Until the `igcdLLL` or similar is added, there is not support of Diophantine solutions for a set of linear equations.

Reply:
Oh, I missed the integer assumptions!